### PR TITLE
[loadconfig] Load model and dataset in config at start

### LINKED ISSUE #389: buf size is determined after minibatch size is set at load config
+ batch seems not updated when loadConfig. please check this as well

### PR DESCRIPTION
**Pull request by kparichay**
Load model and dataset in config at start so that any dependencies
which model, dataset and layer have among them can be easily solved

Resolves #389

**Self evaluation:**
1. Build test: [x]Passed [ ]Failed [ ]Skipped
2. Run test: [x]Passed [ ]Failed [ ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

cc. @zhoonit

### CODE REVIEW

**Review by zhoonit — COMMENTED** [review]
So this fixes the order of [model] and [dataset], am i correct?

File: nntrainer/src/model_loader.cpp [hunk 1]
```diff
@@ -107,11 +117,12 @@ int ModelLoader::loadDatasetConfigIni(dictionary *ini, NeuralNetwork &model) {
   status = parse_and_set("Dataset:LabelData", DATA_LABEL, true);
   NN_RETURN_STATUS();
 
-  /// fixme: #299, #389
   int bufsize = iniparser_getint(ini, "DataSet:BufferSize", model.batch_size);
   ml_logd("buf size: %d", bufsize);
   status = model.data_buffer->setBufSize(bufsize);
   NN_RETURN_STATUS();
+  status = model.data_buffer->setBatchSize(model.batch_size);
```
Comment (zhoonit) [inline nntrainer/src/model_loader.cpp:124]: This should be done before :122

**Review by zhoonit — COMMENTED** [review]
Could  you uncomment `unittest_nntrainer_modelfile.cpp:186` and `unittest_nntrainer_modelfile.cpp:194` to check it is working fine?

File: nntrainer/src/model_loader.cpp (hunk 1 above)
Comment (zhoonit) [reply nntrainer/src/model_loader.cpp:124]: More Idealy, IMO, setBufSize shouldn't rely one batchSize rather initialize checks if bufsize > batchsize though. I think it is slightly out of topic.

**Comment by kparichay** [comment]
> So this fixes the order of [model] and [dataset], am i correct?

Yes, this is fixed. Also fixes the issue where batch size is used before it is set.

File: nntrainer/src/model_loader.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/src/model_loader.cpp:124]: > This should be done before :122

It is not necessary. bufsize is already set at line 120 and initialized with model.batch_size. This line sets model.data_buffer.batch_size.

**Comment by kparichay** [comment]
> Could you uncomment `unittest_nntrainer_modelfile.cpp:186` and `unittest_nntrainer_modelfile.cpp:194` to check it is working fine?

Enabled the test cases, they pass now.

File: nntrainer/src/model_loader.cpp (hunk 1 above)
Comment (kparichay) [reply nntrainer/src/model_loader.cpp]: Updated.

**Review by zhoonit — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**